### ISSUE TITLE
[BUG] [Linter] supplied configuration in CLI overridden by default file

### ISSUE THREAD

**Issue by willemo** [djlint/djLint#803]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info
 - OS: MacOS 14.2.1
 - Python Version (``3.8.16``)
 - djLint Version (``1.34.1``)
 - template language: Jinja

## Issue
When you have a "default" config file in your project's root folder (e.g. a `pyproject.toml` file) and you provide a **different** config file in your CLI command (e.g. `--configuration .djlintrc`), the former will always override the latter. [According to the docs](https://www.djlint.com/docs/configuration/) this should not be the case:

> Command line args will always override any settings in `pyproject.toml`.

I've found a possible cause of this: in the [`load_project_settings`](https://github.com/djlint/djLint/blob/aed8a971cc09a1c8d797ed2d7f6c0eb8931c22a8/src/djlint/settings.py#L109) function there isn't a return after the provided config path has been loaded, causing the default paths to be checked, which will override the provided config.

## How To Reproduce
Create two config files (`pyproject.toml` and `.djlintrc`) with different configuration (e.g. only add `extend_exclude` to the `.djlintrc` file) and run the following command:

```
djlint --configuration .djlintrc
```

The command will ignore the `extend_exclude` settings from `.djlintrc`.

### `pyproject.toml`
```toml
[tool.djlint]
profile = "jinja"
```

### `.djlintrc`
```json
{
    "profile": "jinja",
    "extend_exclude": "path/to/file.html"
}
```

**Comment by keikoro** [comment]
The paragraph from the [Configuration section](https://www.djlint.com/docs/configuration/) from which you quoted continues:

> Command line args will always override any settings in `pyproject.toml`. **Local project settings will always override global configuration files.**


The [CLI usage section](https://www.djlint.com/docs/getting-started/#cli-usage) defines the `--configuration` option as follows*:
```
  --configuration FILE            Path to global configuration file in
                                  djlint.toml or .djlintrc format
```

I interpret this to mean that
a) command line arguments overrule _settings of the same name_ found in any config files and
b) settings in a local config file override _settings of the same name_ in a global config file.

... Where "local" means at the root of the project the files that djLint is run on sit in.

So if you point to a global file which contains a setting that's also contained in the local file, the local setting will take precedence. E.g. a global setting of `indent=2` will be ignored in favour of a local setting of `indent=4`. If you provide the same setting via command line arguments, e.g. `--indent="6"`, both config file settings will be ignored.

Settings which aren't present elsewhere should be "passed through" either way.

I can use settings in global config files fine, so I'm wondering if your problem is more to do with `extend_exclude` specifically than how settings are processed more generally. See e.g. #664 for earlier reports of problems with it. 

*One thing the docs/CLI help seem to have got wrong is that `FILE` can also be a `pyproject.toml` file.

**Comment by monosans** [comment]
I agree that this is not intuitive and does not match the way other familiar tools work. However, this is a breaking change, so I believe it will only be implemented in version 2.0.

**Comment by monosans** [comment]
Reproduced, and the cause is the one you identified. `load_project_settings` loaded the file named by `--configuration` first and then let the discovered project file update over it, so the named file lost on any key the two both set:

```console
$ cat pyproject.toml      # [tool.djlint] indent = 2
$ cat global.djlintrc     # {"indent": 8, "extend_exclude": "sub"}
$ djlint . --configuration global.djlintrc
```

`indent` came out as 2, from `pyproject.toml`. A key only one file sets, such as your `extend_exclude`, passed through either way, which is why the failure looks intermittent depending on which settings overlap.

Changing the default is the breaking change I described, so in v1.45.0 this is opt-in as `--prefer-configuration`:

```console
$ djlint . --configuration global.djlintrc --prefer-configuration
```

`indent` is then 8. Without the flag nothing changes, so existing setups keep the behaviour they have.

The two files still layer: preferring the named file puts it on top of the project's rather than replacing it, so a setting only `pyproject.toml` has still applies. The flag is command line only, since it decides how the configuration files themselves are read.

Leaving this open for the default to flip in 2.0, which is the part that needs a major version.
